WordPress/Classes/ViewRelated/Post/PostListViewController.swift

```diff
@@ -0,0 +1,685 @@
+        let blog = apost.blog
+        
+        if blog.supports(.Stats) {
+            // Needs Jetpack.
+            return
+        }
+        
+        WPAnalytics.track(.PostListStatsAction, withProperties: propertiesForAnalytics())
+        
+        // Push the Stats Post Details ViewController
+        let identifier = NSStringFromClass(StatsPostDetailsTableViewController.self)
+        let service = BlogService(managedObjectContext: ContextManager.sharedInstance().mainContext)
+        let statsBundle = NSBundle(forClass: WPStatsViewController.self)
+        let path = statsBundle.pathForResource("WordPressCom-Stats-iOS", ofType: "bundle")!
+        let bundle = NSBundle(path: path)
+        let statsStoryboard = UIStoryboard(name: self.dynamicType.statsStoryboardName, bundle: bundle)
+        let controller = statsStoryboard.instantiateViewControllerWithIdentifier(identifier) as! StatsPostDetailsTableViewController
+        
+        controller.postID = apost.postID
+        controller.postTitle = apost.titleForDisplay()
+        controller.statsService = WPStatsService(siteId: blog.dotComID, siteTimeZone: service.timeZoneForBlog(blog), oauth2Token: blog.authToken, andCacheExpirationInterval: self.dynamicType.statsCacheInterval)
+        
+        navigationController?.pushViewController(controller, animated: true)
+    }
+    
+    // MARK: - Filter Related
+    
+    override internal func currentPostAuthorFilter() -> PostAuthorFilter {
+        if !canFilterByAuthor() {
+            // No REST API, so we have to use XMLRPC and can't filter results by author.
+            return .Everyone
+        }
+        
+        if let filter = NSUserDefaults.standardUserDefaults().objectForKey(self.dynamicType.currentPostAuthorFilterKey) {
+            if filter.unsignedIntegerValue == PostAuthorFilter.Everyone.rawValue {
+                return .Everyone
+            }
+        }
+        
+        return .Mine
+    }
+
+    override internal func setCurrentPostAuthorFilter(filter: PostAuthorFilter) {
+        guard filter != currentPostAuthorFilter() else {
+            return
+        }
+        
+        WPAnalytics.track(.PostListAuthorFilterChanged, withProperties: propertiesForAnalytics())
+        
+        NSUserDefaults.standardUserDefaults().setObject(filter.rawValue, forKey: self.dynamicType.currentPostAuthorFilterKey)
+        NSUserDefaults.resetStandardUserDefaults()
+        
+        recentlyTrashedPostObjectIDs?.removeAllObjects()
+        resetTableViewContentOffset()
+        updateAndPerformFetchRequestRefreshingCachedRowHeights()
+        syncItemsWithUserInteraction(false)
+    }
+
+    private func keyForCurrentListStatusFilter() -> String {
+        return self.dynamicType.currentPostListStatusFilterKey
+    }
+
+    // MARK: - Cell Delegate Methods
+
+    func cell(cell: UITableViewCell!, receivedEditActionForProvider contentProvider: WPPostContentViewProvider!) {
+        let apost = contentProvider as! AbstractPost
```

Comment:
So, I spent some time thinking about this yesterday and today, and depending on the case I will take on of two actions:

1. In scenarios such as this one, in which the flow of execution would have no impact on the user experience, I will add an assertion for debug, a log for release, and interrupt the method execution.
2. In scenarios in which the invalid internal state would degrade user experience in other ways (such as seeing empty cells), I'll use `precondition()` instead of `assert()`, the rationale being that we really want to avoid invalid bug reports (ie: "I see empty cells") and instead get a crash and a precise stack trace.